ui.apps/src/main/content/jcr_root/apps/aem-dictionary-translator/clientlibs/dictionary/.content.xml

```diff
@@ -1,4 +1,5 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <jcr:root xmlns:cq="http://www.day.com/jcr/cq/1.0" xmlns:jcr="http://www.jcp.org/jcr/1.0"
           jcr:primaryType="cq:ClientLibraryFolder"
-          categories="[dictionary-translator.dictionary]"/>
+          categories="[dictionary-translator.dictionary]"
+          jsProcessor="[default:none,min:gcc;languageIn=ECMASCRIPT_2020;languageOut=ECMASCRIPT_2020]"/>
```

Comment:
This requires 6.5.16 or newer. If it works with ECMASCRIPT_2018 that would be even better (as that is supported since AEM SDK 2020.x and AEM 6.5.0)

Reply:
Good point. In order to have ECMASCRIPT_2018 as highest version we need to rework the optional chaining. But the JavaScript version should not rely on AEM but browser support imo and all major browsers support ECMASCRIPT_2020. AEM 6.5 is not supported as there is the OOTB translator for that and we have some cloud only functionality here. 

@royteeuwen wdyt?

Reply:
For me a TouchUI based translator would be beneficial even for AEM 6.5…

Reply:
This is all about minimization happening on server-side. Newer ECMA script is just not supported by the old Google Closure Compiler used by AEM

Reply:
Yes I agree but that will require some effort as it was "designed" cloud only

Reply:
For me personally I of course don't mind if it works on AEM 6.5, it would be nice to get rid of that last classic ui console ;), but I don't have any clients where it would be used so it would be "best effort" and if someone has any errors that it doesn't work on AEM 6.5 that they contribute the fixes.

Did you already test it on AEM 6.5 @kwin ? Would be nice to hear feedback. If this ecmascript is the only thing, then thats an easy fix and we can easily do this

Reply:
I will make sure the javascript will compile with `ECMASCRIPT_2018`

Reply:
For me it is fair to require 6.5.16 or above

Reply:
I also quickly checked AEM 6.5 support and it's pretty easy to do now we are using the endpoint for replication ;) I will try to create a pull request for this ;)

Reply:
fyi: AEM 6.5.17 contains this version: https://github.com/google/closure-compiler/wiki/releases#january-6-2021-v20210106, not sure which AEM SDK version this shipped with but in has support for ECMASCRIPT_2020 ;) but will set to 2018 just to be sure

Reply:
FTR, this is what I once found out:
AEM Version | GCC Version | Supported Languages | Source | Comment
-- | -- | -- | -- | --
AEM 6.4 | v20170806 | ECMASCRIPT3,  ECMASCRIPT5,  ECMASCRIPT5_STRICT,  ECMASCRIPT_2015,  ECMASCRIPT6_TYPED,  ECMASCRIPT_2016,  ECMASCRIPT_2017,  ECMASCRIPT_NEXT,  NO_TRANSPILE (only valid for LanguageOut) | https://github.com/google/closure-compiler/blob/733fbd4c3b859ce9ed119cecb31cd07b429dbf4d/src/com/google/javascript/jscomp/CompilerOptions.java#L3028 | Partial ES6 support for output: https://github.com/google/closure-compiler/wiki/Releases#june-26-2017-v20170626
AEM 6.5 and AEM SDK 202010 | v20190121 | ECMASCRIPT3,  ECMASCRIPT5,  ECMASCRIPT5_STRICT,  ECMASCRIPT_2015,  ECMASCRIPT6_TYPED,  ECMASCRIPT_2016,  ECMASCRIPT_2017,  ECMASCRIPT_2018, ECMASCRIPT_NEXT,  NO_TRANSPILE (only valid for LanguageOut) | https://github.com/google/closure-compiler/blob/5e61881d48a6eef9110369f235d1ac53e9f6d23f/src/com/google/javascript/jscomp/CompilerOptions.java#L3084 | Almost complete ES6 support for output.
AEM 6.5.16 and AEM SDK 2023.1.10912.20230130T173736Z  | v20210106 | ECMASCRIPT3,  ECMASCRIPT5,  ECMASCRIPT5_STRICT,  ECMASCRIPT_2015,  ECMASCRIPT6_TYPED,  ECMASCRIPT_2016,  ECMASCRIPT_2017,  ECMASCRIPT_2018,ECMASCRIPT_2019,ECMASCRIPT_2020,ECMASCRIPT_NEXT,  NO_TRANSPILE (only valid for LanguageOut) | https://github.com/google/closure-compiler/blob/6f674dccb0fb8f6c1b930cf14beb38f43f81338c/src/com/google/javascript/jscomp/CompilerOptions.java#L2859

Reply:
allright, we will be supporting 6.5.17 and 2023.1.10912.20230130T173736Z as minimal version ;) I double checked the bundle and the info is correct, I will be adding this in #66